### ISSUE TITLE
0.0.40 regression: `@cached_property` returning Callable[...] flagged as invalid-assignment against metaclass-declared Callable attribute

### ISSUE THREAD

**Issue by adamtheturtle** [astral-sh/ty#3568]
### Summary

In ty 0.0.40, a `@cached_property` returning `Callable[...]` on a class whose metaclass declares the same attribute as `Callable[...]` is newly flagged with `invalid-assignment`. ty 0.0.39 accepted this; pyright, pyrefly, and mypy all accept it.

### Repro (no external deps)

`repro.py`:
```python
from collections.abc import Callable
from functools import cached_property


class LanguageCls(type):
    format_call_arg: Callable[[str, str], str]


def passthrough(value: str, rendered: str) -> str:
    return rendered


class Rust(metaclass=LanguageCls):
    @cached_property
    def format_call_arg(self) -> Callable[[str, str], str]:
        return passthrough
```

Run:
```bash
uvx ty==0.0.40 check repro.py
```

### Actual result

```
error[invalid-assignment]: Object of type `cached_property[(str, str, /) -> str]` is not assignable to attribute `format_call_arg` of type `(str, str, /) -> str`
  --> repro.py:14:9
   |
14 |     def format_call_arg(self) -> Callable[[str, str], str]:
   |         ^^^^^^^^^^^^^^^
```

### Expected result

No diagnostic. The `cached_property` descriptor yields the cached `Callable[[str, str], str]` on access, matching the metaclass-declared attribute type.

### Cross-checker results

| Checker | Result |
|---|---|
| `uvx ty==0.0.40 check repro.py` | **ERROR (regression)** |
| `uvx ty==0.0.39 check repro.py` | clean |
| `uvx pyright==1.1.409 repro.py` | clean |
| `uvx pyrefly==1.0.0 check repro.py` | clean |
| `uvx mypy==2.1.0 --strict --config-file=/dev/null repro.py` | clean |

### Notes

Closed #1446 originally added `cached_property` descriptor handling, which 0.0.40 appears to have regressed for this shape. #3256 covers a different cached_property case (enclosing-class `TypeVar` in the return type); this repro is non-generic.

**Comment by charliermarsh** [comment]
Accepting that seems... unsound?
```python
from collections.abc import Callable
from functools import cached_property


class LanguageCls(type):
    format_call_arg: Callable[[str, str], str]


def passthrough(value: str, rendered: str) -> str:
    return rendered


class Rust(metaclass=LanguageCls):
    @cached_property
    def format_call_arg(self) -> Callable[[str, str], str]:
        return passthrough


def render(cls: LanguageCls) -> str:
    return cls.format_call_arg("value", "rendered")


render(Rust)  # runtime failure: cached_property is not callable
```

**Comment by charliermarsh** [comment]
I'm going to close for now, but someone else may come along and correct me :)
